Snapshot for `-profile test_assembly_input`

dialvarezs opened the pull request:
<!--
# nf-core/mag pull request

Many thanks for contributing to nf-core/mag!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/mag/tree/main/.github/CONTRIBUTING.md)
-->

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/main/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

dialvarezs commented:
I also updated the `get_mag_depths.py` script to produce depth summaries with consistent column ordering, to prevent swapped depth values on the merged depth summary.

d4straub commented on conf/modules.config:
```diff
@@ -463,12 +463,13 @@ process {
         publishDir = [
             [
-                path: { "${params.outdir}/GenomeBinning/QC/BUSCO/${meta.id}" },
+                path: { "${params.outdir}/GenomeBinning/QC/BUSCO/${meta.assembler}-${meta.binner}-${meta.domain}-${meta.refinement}-${meta.id}" },
```
Will that need an update of the output documentation?

d4straub commented on tests/test_assembly_input.nf.test:
```diff
@@ -15,6 +15,63 @@ nextflow_pipeline {
+            def stable_name_binning = getAllFilesFromDir(
+                params.outdir,
+                include: ['GenomeBinning/**'],
+                ignore: ['GenomeBinning/QC/BUSCO/**/{logs,prodigal_output,busco_sequences}/**'],
```
doesnt that overlap with .nftignore? Wouldnt it be better to use that file instead?

dialvarezs commented:
I added some basic CSV checks for `busco_summary` and `bin_summary` files. I also updated the nft-utils plugin.
And fixed the error reporting of tiara_classify that was breaking the `test_alternatives` test.

Most of the small fixes I did on this PR originally were backported to other PR for v5.2.0, so this PR has become more simple now.

prototaxites commented on tests/test_assembly_input.nf.test:
```diff
@@ -27,12 +86,48 @@ nextflow_pipeline {
+                { assert busco_logs.collect {
+                        log -> log.readLines().any { it.contains('BUSCO analysis done.') }
+                    }.every()
+                },
```
Does BUSCO ever fail on any of the bins in the tests?

prototaxites reviewed (approved):
A few questions/comments, but this is looking fine as-is!

dialvarezs commented on tests/test_alternatives.nf.test:
```diff
@@ -72,13 +72,12 @@ nextflow_pipeline {
-                        removeNextflowVersion("${outputDir}/pipeline_info/nf_core_mag_software_mqc_versions.yml"),
-                    ).match()
-                },
+                    workflow.trace.succeeded().size(),
```
Yes, it works with both 25.04 and 25.10. It seems that the issue was with a pre-release version (25.07)

dialvarezs commented on tests/test_assembly_input.nf.test:
```diff
@@ -27,12 +86,48 @@ nextflow_pipeline {
+                { assert dastool_unbins.collect { file -> path("${file}").fasta.size() >= 0 }.every() },
+
+                // CSV checks
+                { assert path(bin_summary).csv(sep: "\t").rowCount > 0 },
```
In this case we're only using the refined bins so it would be = to dastool bins. I'll add that.
